test/BlankNodeManagerTest.cpp

```diff
@@ -9,6 +15,142 @@
+// Test fixture providing common infrastructure for BlankNodeManager tests
+class BlankNodeManagerTestFixture : public ::testing::Test {
+ protected:
+  // Helper to create a BlankNodeManager with default minIndex
+  static std::unique_ptr<BlankNodeManager> createManager(
+      uint64_t minIndex = 0) {
+    return std::make_unique<BlankNodeManager>(minIndex);
+  }
+
+  // Helper to create a LocalBlankNodeManager
+  static std::shared_ptr<BlankNodeManager::LocalBlankNodeManager>
+  createLocalManager(BlankNodeManager* bnm) {
+    return std::make_shared<BlankNodeManager::LocalBlankNodeManager>(bnm);
+  }
+
+  // Helper to get the primary blocks from a LocalBlankNodeManager
+  static auto& getPrimaryBlocks(BlankNodeManager::LocalBlankNodeManager& lbnm) {
+    return lbnm.blocks_->blocks_;
+  }
+
+  // Helper to get the total number of blocks (primary + other)
+  static size_t getTotalBlockCount(
+      const BlankNodeManager::LocalBlankNodeManager& lbnm) {
+    size_t count = lbnm.blocks_->blocks_.size();
+    for (const auto& otherBlocks : lbnm.otherBlocks_) {
+      count += otherBlocks->blocks_.size();
+    }
+    return count;
+  }
+
+  // Helper to get all block indices (from both primary and other blocks)
+  static std::vector<uint64_t> getAllBlockIndices(
+      const BlankNodeManager::LocalBlankNodeManager& lbnm) {
+    std::vector<uint64_t> indices;
+    for (const auto& block : lbnm.blocks_->blocks_) {
+      indices.push_back(block.blockIdx_);
+    }
+    for (const auto& otherBlocks : lbnm.otherBlocks_) {
+      for (const auto& block : otherBlocks->blocks_) {
+        indices.push_back(block.blockIdx_);
+      }
+    }
+    return indices;
+  }
+
+  // Helper to allocate N IDs from a LocalBlankNodeManager
+  static std::vector<uint64_t> allocateIds(
+      BlankNodeManager::LocalBlankNodeManager& lbnm, size_t count) {
+    std::vector<uint64_t> ids;
+    ids.reserve(count);
+    for (size_t i = 0; i < count; ++i) {
+      ids.push_back(lbnm.getId());
+    }
+    return ids;
+  }
+
+  // Helper to allocate IDs that span multiple blocks
+  static std::vector<uint64_t> allocateIdsAcrossBlocks(
+      BlankNodeManager::LocalBlankNodeManager& lbnm, size_t numBlocks) {
+    std::vector<uint64_t> ids;
+    for (size_t i = 0; i < numBlocks; ++i) {
+      // Allocate at least one ID per block
+      ids.push_back(lbnm.getId());
+      // Fill the rest of the block to force allocation of next block
+      if (i < numBlocks - 1) {
+        auto& blocks = getPrimaryBlocks(lbnm);
+        if (!blocks.empty()) {
+          blocks.back().nextIdx_ =
+              blocks.back().startIdx_ + BlankNodeManager::blockSize_;
+        }
+      }
+    }
+    return ids;
+  }
+
+  // Helper to verify all IDs are contained in the LocalBlankNodeManager
+  static void verifyIdsContained(
+      const BlankNodeManager::LocalBlankNodeManager& lbnm,
+      const std::vector<uint64_t>& ids) {
```

Comment:
Consider adding a source location + location trace here. Same for all the other functions containing `EXPECT`/`ASSERT`.